File: lacumbre/model/privacy.go
package model

import "time"

type Privacy struct {
	UserID                string    `gorm:"primaryKey" json:"user_id"`
	Email                 string    `json:"email"`
	PhoneNumber           string    `json:"phone_number"`
	Pronouns              string    `json:"pronouns"`
	Gender                string    `json:"gender"`
	Location              string    `json:"location"`
	Status                string    `json:"status"`
	PushNotifications     string    `json:"push_notifications"`
	PushNotificationToken string    `json:"push_notification_token"`
	ScheduleReminders     string    `json:"schedule_reminders"`
	PasstimeReminders     string    `json:"passtime_reminders"`
	FinalReminders        string    `json:"final_reminders"`
	UpdatedAt             time.Time `gorm:"autoUpdateTime" json:"updated_at"`
	CreatedAt             time.Time `gorm:"autoCreateTime" json:"created_at"`
}

func (Privacy) TableName() string {
	return "user_privacy"
}

File: lib/models/privacy.dart
class Privacy {
  String userID = "";
  String email = "PUBLIC";
  String phoneNumber = "FRIENDS";
  String pronouns = "PRIVATE";
  String gender = "PRIVATE";
  String location = "DISABLED";
  String status = "PUBLIC";
  String pushNotifications = "DISABLED";
  String pushNotificationToken = "";
  String scheduleReminders = "";
  DateTime updatedAt = DateTime.now().toUtc();
  DateTime createdAt = DateTime.now().toUtc();

  Privacy();

  Privacy.fromJson(Map<String, dynamic> json) {
    userID = json["user_id"] ?? "";
    email = json["email"] ?? "PUBLIC";
    phoneNumber = json["phone_number"] ?? "FRIENDS";
    pronouns = json["pronouns"] ?? "PRIVATE";
    gender = json["gender"] ?? "PRIVATE";
    location = json["location"] ?? "DISABLED";
    status = json["status"] ?? "PUBLIC";
    pushNotifications = json["push_notifications"] ?? "DISABLED";
    pushNotificationToken = json["push_notification_token"] ?? "";
    scheduleReminders = json["schedule_reminders"] ?? "";
    updatedAt = DateTime.tryParse(json["updated_at"]) ?? DateTime.now().toUtc();
    createdAt = DateTime.tryParse(json["created_at"]) ?? DateTime.now().toUtc();
  }

  Map<String, dynamic> toJson() {
    return {
      "user_id": userID,
      "email": email,
      "phone_number": phoneNumber,
      "pronouns": pronouns,
      "gender": gender,
      "location": location,
      "status": status,
      "push_notifications": pushNotifications,
      "push_notification_token": pushNotificationToken,
      "schedule_reminders": scheduleReminders,
      "updated_at": updatedAt.toIso8601String(),
      "created_at": createdAt.toIso8601String()
    };
  }
}

/*
{
    "user_id": "bye",
    "email": "PUBLIC",
    "phone_number": "PRIVATE",
    "pronouns": "",
    "gender": "",
    "location": "DISABLED",
    "push_notifications": "ENABLED",
    "push_notification_token": "FCM-TOKEN-1239",
    "updated_at": "2022-05-27T00:25:26.862199Z",
    "created_at": "2022-05-27T00:16:17.483235Z"
}
 */